temporian/core/evaluation.py

```diff
@@ -222,25 +223,50 @@ def build_schedule(
     for op in graph.operators:
         num_pending_inputs = 0
         for input_node in op.inputs.values():
+            node_to_op[input_node].append(op)
             if input_node in graph.inputs:
                 # This input is already available
                 continue
-            node_to_op[input_node].append(op)
             num_pending_inputs += 1
         if num_pending_inputs == 0:
             # Ready to be scheduled
-            ready_ops.add(op)
+            ready_ops.append(op)
+            ready_ops_set.add(op)
         else:
             # Some of the inputs are missing.
             op_to_num_pending_inputs[op] = num_pending_inputs
 
+    # Make evaluation order deterministic.
+    #
+    # Execute the op with smallest internal ordered id first.
+    ready_ops.sort(key=lambda op: op._internal_ordered_id, reverse=True)
```

Comment:
I'm not sure why the scheduling order is better for memory release with this step.
I get it only affects the ops that depend directly on the inputs (current ops in `ready_ops`) and executes them in instantiation order, but is that relevant in terms of memory allocation?